check_limits.py

```diff
@@ -1,18 +1,170 @@
+# =============================================================================
+import random
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
+#Declare global variables
+CURRENT_BATTERYPACK   =   'Lithium'
+LANGUAGE              =   'EN'
+ALERTS                =    []
 
-  return True
+#Define multi language error messages
+error_messages               = {'low_breach'  : { 'DE' : 'Untergrenze überschritten für ', 
+                                                  'EN' : 'Lower Limit Breached for ' }   ,
+                                'low_warning' : { 'DE' : 'Warnung vor Untergrenze für '  ,   
+                                                  'EN' : 'Warning: Lower Limit approaching for '},
+                                'high_breach' : { 'DE' : 'Obergrenze überschritten für '  , 
+                                                  'EN' : 'Higher Limit Breached for ' }   ,
+                                'high_warning': { 'DE' : 'Warnung vor höherer Grenze für ', 
+                                                  'EN' : 'Warning: Higher Limit approaching for ' } }
 
+#MVC Model
+#Declare operating threshold params for different battery packs
+batteryThresholdParams = { 'Lithium' : 
+                            {
+                            'Temperature'  : { 'lower':  0   , 'upper': 45  },
+                            'StateOfCharge': { 'lower': 20   , 'upper': 80  },
+                            'ChargeRate'   : { 'lower': 0    , 'upper': 0.8 },
+                            }, 
+                                    
+                         'NiMh' :   
+                            {
+                            'Temperature'  : { 'lower': -20  , 'upper': 40 },
+                            'StateOfCharge': { 'lower':  20  , 'upper': 80  },
+                            'ChargeRate'   : { 'lower': 0    , 'upper': 0.8 },
+                            },   
+                         }
+
+def getBatteryThresholdLimit(batteryType):
+    return batteryThresholdParams[batteryType]
+
+#Check if Battery is working fine
+def battery_is_ok(**kwargs):
+  
+  batteryLimits             = getBatteryThresholdLimit(CURRENT_BATTERYPACK)
+    
+  global BATTERY_CONDITION_ALL_OK
+  BATTERY_CONDITION_ALL_OK  = True
+  
+  for criteria, criteriavalue in kwargs.items():
+      lower,upper    = getBoundaryConditions(batteryLimits,criteria, criteriavalue)
+      lower_status   = checkLowerLimitBreach(criteriavalue,lower,upper) 
+      upper_status   = checkUpperLimitBreach(criteriavalue,upper)
+      hasBreached    = checkBreaches(lower_status, upper_status)
+
+      if hasBreached:
+          BATTERY_CONDITION_ALL_OK = False
+          global ALERTS
```

Comment:
If you make the ALERTS.append inside this function, setErrorMessages does not need to do that.
Then you can avoid making ALERTS global, helping in thread-safety.